### PR TITLE
refactor: add FS Router

### PR DESCRIPTION
**Pull request by Lodin**
Requires vaadin/hilla#2214 to be released

### CODE REVIEW

File: src/main/frontend/themes/skeleton-starter-hilla-react/styles.css
Comment (platosha) [inline src/main/frontend/themes/skeleton-starter-hilla-react/styles.css:1]: What's the intention for this file, is it just an example? If so, I'd suggest to remove it.

File: src/main/frontend/views/$layout.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { ProgressBar } from '@vaadin/react-components/ProgressBar';
+import { Suspense, useEffect } from 'react';
+import { NavLink, Outlet } from 'react-router-dom';
+
+const navLinkClasses = ({ isActive }: any) => {
+  return `block rounded-m p-s ${isActive ? 'bg-primary-10 text-primary' : 'text-body'}`;
+};
+
+export default function MainLayout() {
+  const currentTitle = useViewConfig()?.title ?? 'Hybrid example';
```
Comment (platosha) [inline src/main/frontend/views/$layout.tsx]: What about reading the default title from HTML?



```suggestion

const defaultTitle = document.title;

export default function MainLayout() {
  const currentTitle = useViewConfig()?.title ?? defaultTitle;
```

**Review by platosha — CHANGES_REQUESTED** [review]

File: src/main/frontend/themes/skeleton-starter-hilla-react/styles.css
Comment (Lodin) [reply src/main/frontend/themes/skeleton-starter-hilla-react/styles.css:1]: Well, actually, it existed as an empty file, so I just have added a some styles to make it prettier (without that style, the text field at the route page sticks to the main menu).

**Review by platosha — APPROVED** [review]

**Review by platosha — CHANGES_REQUESTED** [review]
Let's wait for Hilla 24.4.0.alpha16 before merging

**Review by platosha — APPROVED** [review]

**Pull request merged**